planner/core/preprocess.go

```diff
@@ -570,6 +604,14 @@ func (p *preprocessor) Leave(in ast.Node) (out ast.Node, ok bool) {
 		if cteNode, exist := x.(*ast.CommonTableExpression); exist {
 			with.cteCanUsed = append(with.cteCanUsed, cteNode.Name.L)
 		}
+	case *ast.SelectStmt:
+		if x.With != nil {
+			p.preprocessWith.cteStack = p.preprocessWith.cteStack[0 : len(p.preprocessWith.cteStack)-1]
```

Comment:
`[: len(p.preprocessWith.cteStack)-1]` can be this for simplicity

Reply:
Great ~ . I will fix it in the next time ~